modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -946,6 +946,12 @@ else if (!deleteRelationshipObjectField) {
 			objectField1 -> !objectField1.isMetadata());
 
 		if (objectDefinition.isApproved() && (objectFields.size() == 1)) {
+			if (deleteRelationshipObjectField) {
+				throw new RequiredObjectFieldException.
+					MustNotDeleteObjectFieldRelationship(
+						objectDefinition.getShortName(), objectField.getName());
+			}
+
 			throw new RequiredObjectFieldException.
 				MustNotDeleteObjectFieldPublishedObjectDefinition(
 					objectField.getName());
```

Comment:
@larissacribeiro can we change the order to sort alphabetically by exception name?

```
if (!deleteRelationshipObjectField) {
	throw new RequiredObjectFieldException.
		MustNotDeleteObjectFieldPublishedObjectDefinition(
			objectField.getName());
}
else {
	throw new RequiredObjectFieldException.
		MustNotDeleteObjectFieldRelationship.
			objectDefinition.getShortName(), objectField.getName());
}
```

Reply:
yes, @carolmariaabb, just did! thanks!